### PR TITLE
JAMES-3198: UserMailboxes api should provide detailMessage when error happened

### PR DESCRIPTION
**Pull request by nvduc91**
UserMailboxes api should provide detailMessage when error happened

### CODE REVIEW

File: server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserMailboxesRoutes.java [hunk 1]
```diff
@@ -281,6 +281,7 @@ public void defineMailboxExists() {
                         .statusCode(HttpStatus.NOT_FOUND_404)
                         .type(ErrorType.INVALID_ARGUMENT)
                         .message("Invalid get on user mailboxes")
+                        .cause(new IllegalArgumentException("Mailboxes do not exist"))
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserMailboxesRoutes.java:284]: 1. It is `Mailbox does not exist`

2. I'm against creating an exception that we did not encountered.

I suggest:

```
throw ErrorResponder.builder()
                        .statusCode(HttpStatus.NOT_FOUND_404)
                        .type(ErrorType.INVALID_ARGUMENT)
                        .message("Mailbox does not exist")
                        .haltError();
```

File: server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/UserMailboxesService.java [hunk 2]
```diff
@@ -120,7 +120,7 @@ private void deleteMailbox(MailboxSession mailboxSession, MailboxPath mailboxPat
     }
 
     private void usernamePreconditions(Username username) throws UsersRepositoryException {
-        Preconditions.checkState(usersRepository.contains(username));
+        Preconditions.checkState(usersRepository.contains(username), "User do not exist");
```
Comment (chibenwa) [inline server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/UserMailboxesService.java:123]: `User does not exist`

Comment (nvduc91) [reply server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/UserMailboxesService.java:123]: agree

File: server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserMailboxesRoutes.java (hunk 1 above)
Comment (nvduc91) [reply server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserMailboxesRoutes.java:284]: It seem not right: we need provide more `details` error but not skip it or replace it in `message`.
I dont know any better way to do it yet.

Comment (chibenwa) [reply server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserMailboxesRoutes.java]: > we need provide more details

No not really, it is a suggestion from the QA team.

`.message("Invalid get on user mailboxes")` do not means anything to me, that message is useless, we'd rather replace it with our usefull message.

> you might break others test with it.

Which ones?

Comment (nvduc91) [reply server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserMailboxesRoutes.java]: `deleteShouldRemoveMailbox`

Comment (chibenwa) [reply server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserMailboxesRoutes.java]: Probably the error message is not accurate there too

**Comment by nvduc91** [comment]
I dont know why is happen.
```
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: cassandra-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: elasticSearch-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: rabbitmq-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: swift-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: integration-cassandra-rabbitmq-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: integration-cassandra-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: integration-jpa-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: integration-spring-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: integration-memory-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: james-parent-compile-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: james-parent-test-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: james-server-cassandra-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: james-server-cassandra-rabbitmq-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: james-server-cassandra-ldap-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: james-server-jpa-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: james-server-demo-testing1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: james-server-spring-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: james-server-memory-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: No such container: git-publish-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] + true
[19d2c315f975308bf0b07b837380adaa55a9864a] Running: Shell Script
[19d2c315f975308bf0b07b837380adaa55a9864a] [173] Running shell script
[19d2c315f975308bf0b07b837380adaa55a9864a] + docker volume rm m2-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd
[19d2c315f975308bf0b07b837380adaa55a9864a] Error response from daemon: get m2-1378-ed97cf90-3ff0-4137-bb68-ce73731a28cd: no such volume
```

**Comment by nvduc91** [comment]
test this please

**Comment by chibenwa** [comment]
```
[19d2c315f975308bf0b07b837380adaa55a9864a] + git checkout 1378
[19d2c315f975308bf0b07b837380adaa55a9864a] fatal: reference is not a tree: 1378
```

Don't use numbers for branch names. Please rename your branch `JAMES-3198` and reopen the PR

**Pull request closed without merging**